src/core/pal/feature.cpp

```diff
@@ -1170,6 +1170,19 @@ int FeaturePart::createCurvedCandidatesAlongLine( QList< LabelPosition* >& lPos,
     return 0;
   }
 
+  //calculate overall angle of line
+  double lineAngle;
```

Comment:
The block calculates the angle of the complete label instead of just the first character. It better determines if a label is upsidedown.
But lineAngle is not used now. It is part of a former commit and is not part of the bug fix. You are correct to remove it now. I'll address it later.